services/QuillLMS/app/lib/rule_feedback_history.rb

```diff
@@ -40,8 +44,10 @@ def self.feedback_history_to_json(f_h)
         }
     end
 
-    def self.generate_rulewise_report(rule_uid:, prompt_id: )
-        feedback_histories = FeedbackHistory.where(rule_uid: rule_uid, prompt_id: prompt_id, used: true)
+    def self.generate_rulewise_report(rule_uid:, prompt_id:, start_date:, end_date:)
+        start_date_param = start_date || FeedbackHistory.first&.created_at
```

Comment:
nit: you can avoid a DB call by just using a really early timestamp, like Jan 1 2000